### ISSUE TITLE
Bundler: optional groups break license_finder

### ISSUE THREAD

**Issue by steakknife** [pivotal-legacy/LicenseFinder#493]
### Gemfile

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'

gem 'license_finder'

group :xml, optional: true do
  gem 'nokogiri', require: nil
end
```

### Output

```
$ bundle
...
$ license_finder
LicenseFinder::Bundler: is active
~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:40:in `block in for': Unable to find a spec satisfying nokogiri in the set. Perhaps the lockfile is corrupted? (RuntimeError)
	from ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:25:in `loop'
	from ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:25:in `for'
	from ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/definition.rb:237:in `specs_for'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:52:in `gem_details'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:42:in `details'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:13:in `current_packages'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_manager.rb:90:in `current_packages_with_relations'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `each'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `flat_map'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `active_packages'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/core.rb:81:in `current_packages'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/core.rb:76:in `decision_applier'
	from /opt/rubies/ruby-2.5.1/lib/ruby/2.5.0/forwardable.rb:223:in `any_packages?'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:15:in `block in any_packages?'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:13:in `map'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:13:in `any_packages?'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/cli/main.rb:88:in `action_items'
	from ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/bin/license_finder:5:in `<top (required)>'
	from ~/.gem/ruby/2.5.1/bin/license_finder:23:in `load'
	from ~/.gem/ruby/2.5.1/bin/license_finder:23:in `<main>'
$
```

### Retrying using `bundle exec ...`

`bundle exec license_finder` also explodes

```
$ bundle exec license_finder
LicenseFinder::Bundler: [32mis active[0m
bundler: failed to load command: license_finder (~/.gem/ruby/2.5.1/bin/license_finder)
RuntimeError: Unable to find a spec satisfying nokogiri in the set. Perhaps the lockfile is corrupted?
  ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:40:in `block in for'
  ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:25:in `loop'
  ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/spec_set.rb:25:in `for'
  ~/.gem/ruby/2.5.1/gems/bundler-1.16.1/lib/bundler/definition.rb:237:in `specs_for'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:52:in `gem_details'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:42:in `details'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_managers/bundler.rb:13:in `current_packages'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/package_manager.rb:90:in `current_packages_with_relations'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `each'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `flat_map'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/scanner.rb:15:in `active_packages'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/core.rb:81:in `current_packages'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/core.rb:76:in `decision_applier'
  /opt/rubies/ruby-2.5.1/lib/ruby/2.5.0/forwardable.rb:223:in `any_packages?'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:15:in `block in any_packages?'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:13:in `map'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/license_aggregator.rb:13:in `any_packages?'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/lib/license_finder/cli/main.rb:88:in `action_items'
  ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
  ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
  ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
  ~/.gem/ruby/2.5.1/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
  ~/.gem/ruby/2.5.1/gems/license_finder-5.2.3/bin/license_finder:5:in `<top (required)>'
  ~/.gem/ruby/2.5.1/bin/license_finder:23:in `load'
  ~/.gem/ruby/2.5.1/bin/license_finder:23:in `<top (required)>'
$
```

**Comment by kdykeman** [comment]
Hi @steakknife, there is a mechanism within license_finder to explicitly exclude optional groups. In the example above, you would want to do something like:
```
license_finder ignored_groups add xml
```
I have verified that works without errors. I've also verified that if I `bundle install --with xml` license_finder works as expected if xml isn't added to the ignored_groups list.

I suppose license_finder could try to make assumptions about which optional groups are installed by parsing the Gemfile (which isn't currently being parsed) and comparing that to what's in Gemfile.lock, but I think it's best to keep it explicit for now given that the prepare command for bundler doesn't currently deal with optional groups at all.
